lib/src/tasks/coverage/api.dart

```diff
@@ -213,41 +215,50 @@ class CoverageTask extends Task {
   Future _collect() async {
     List<File> collections = [];
     for (int i = 0; i < _files.length; i++) {
-      File collection =
-          new File(path.join(_collections.path, '${_files[i].path}.json'));
-      int observatoryPort;
+      List<int> observatoryPorts;
+      TaskProcess process;
 
       // Run the test and obtain the observatory port for coverage collection.
       try {
-        observatoryPort = await _test(_files[i]);
+        observatoryPorts = await _test(_files[i]);
       } on CoverageTestSuiteException {
         _coverageErrorOutput.add('Tests failed: ${_files[i].path}');
         continue;
       }
 
-      // Collect the coverage from observatory.
-      String executable = 'pub';
-      List args = [
-        'run',
-        'coverage:collect_coverage',
-        '--port=${observatoryPort}',
-        '-o',
-        collection.path
-      ];
+//      for (int j = 0; j < observatoryPorts.length; j++) {
+      for (int j = observatoryPorts.length - 1; j >= 0; j--) {
```

Comment:
Why are you iterating over these ports in reverse? Might be worth adding a comment.
Also get rid of the commented out line.